### PR TITLE
Feature/file data type

### PR DESCRIPTION
**Pull request by TDH55**

### CODE REVIEW

**Comment by TDH55** [comment]
[File data type docs](https://trello.com/c/8Vqy9j5O/86-file-data-type-docs)

File: website/docs/docs/interactions/control-types.md [hunk 1]
```diff
@@ -156,3 +161,19 @@ Here's what the first one looks like in the editor:
 And here's what the second one looks like in the editor:
 
 <img src="/img/autosaveInput-example-lists.png" alt='Autosave Input Example' />
+
+## Modifying Values From a List
+
+Sometimes the values or files that you want to modify in the database may be part of a list. In order to give a prop teh `autosaveInput` or `upload` role, as well as the `listItem` role, you must set the `role` prop to an array containing the two values as opposed to a string of one value. For example, a prop with the roles of `listItem` and `autosaveInput` would look like:
```
Comment (scott-massey) [inline website/docs/docs/interactions/control-types.md:167]:
```suggestion
Sometimes the values or files that you want to modify in the database may be part of a list. In order to give a prop the `autosaveInput` or `upload` role, as well as the `listItem` role, you must set the `role` prop to an array containing the two values as opposed to a string of one value. For example, a prop with the roles of `listItem` and `autosaveInput` would look like: 
```

File: website/docs/api-reference/configuration/manifest-json.md [hunk 2]
```diff
@@ -146,7 +146,7 @@ The name to show in the Adalo editor and displayed to the end user
 - **Required**
 
 The type of value that is expected. These follow Adalo's datatypes. The type will change what gets rendered in the left panel in the editor.
-
+<!-- TODO: add `see more` link for file and image types -->
```
Comment (scott-massey) [inline website/docs/api-reference/configuration/manifest-json.md:149]: is this to-do to-done?

**Review by scott-massey — COMMENTED** [review]
Generally looks great! It would be helpful to also include an example of accessing a new file or image type

File: website/docs/docs/interactions/actions.md [hunk 3]
```diff
@@ -77,3 +77,9 @@ exampleFunc = () => {
   if (exampleAction) exampleAction(arg1, arg2);
 };
 ```
+
+:::note
+
+Sending the values to Adalo varies slightly when dealing with files or images. For more information, see [Files and Images](./files-and-images).
```
Comment (dina-kazakevich) [inline website/docs/docs/interactions/actions.md:83]: invalid link, page not found, not sure if this is intended.

File: website/docs/api-reference/configuration/manifest-json.md [hunk 4]
```diff
@@ -155,6 +155,8 @@ The type of value that is expected. These follow Adalo's datatypes. The type wil
 | `icon`    | The name of a material icon to use. For a full list of possible icon name, see [Material Icons](https://oblador.github.io/react-native-vector-icons/)                                                      |
 | `color`   | A color string: `#ffffff` or `rgba(255, 255, 255, 1)`. See the [colors and branding](/docs/design/colors-branding) reference for more information.                                                         |
```
Comment (dina-kazakevich) [inline website/docs/api-reference/configuration/manifest-json.md:156]: invalid url, page not found, not sure if this is intended

File: website/docs/api-reference/configuration/manifest-json.md [hunk 5]
```diff
@@ -155,6 +155,8 @@ The type of value that is expected. These follow Adalo's datatypes. The type wil
...
 | `date`    | An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) date string: `"2018-01-01T00:00:00.000Z"`                                                                                                            |
+| `file`    | A url to a hosted file. |
+| `image`   | A url to a hosted image. |
 | `list`    | An array of dynamic data. See [Lists](#lists)                                                                                                                                                              |
 | `action`  | A user action such as linking to another page, calling an API, etc. Your component will receive a function returning a promise. The promise will return when all of the associated actions have completed. |
```
Comment (dina-kazakevich) [inline website/docs/api-reference/configuration/manifest-json.md:161]: have **been** completed?

File: website/docs/docs/interactions/files-and-images.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+## Prop Value
+
+Props of type `file` or `image` will pass their values to the component as an object. If the maker has chosen the file or image via the URL option, this object will contain only the location of the hosted image or file and be structured as `{ uri }`. Otherwise, the object will be structured as `{ uri, filename, size }` where `uri` is the location of the hosted file/image, `filename` is the name of the file/image, and `size` is the size of the file/image in bytes. This object can be accessed in the code using `propName.value`.
+
+## `onChange` Functions
+
+When a prop of type `file` or `image` is given the role of either `formValue` or `autosaveInput`, the usage of the resulting `onChange` function will vary from it's usage with other datatypes. When working with these types, the `onChange` function will need to be passed an object containing the `filename` and the `uri` (either local or hosted) or `data` (as a base64 encoded string) of the new file, ex: `{ uri, filename }` or `{ data, filename }`. If the object contains both `data` and `uri`, the `data` field will be used by default.
```
Comment (dina-kazakevich) [inline website/docs/docs/interactions/files-and-images.md:14]:
```suggestion
When a prop of type `file` or `image` is given the role of either `formValue` or `autosaveInput`, the usage of the resulting `onChange` function will vary from its usage with other datatypes. When working with these types, the `onChange` function will need to be passed an object containing the `filename` and the `uri` (either local or hosted) or `data` (as a base64 encoded string) of the new file, ex: `{ uri, filename }` or `{ data, filename }`. If the object contains both `data` and `uri`, the `data` field will be used by default.
```

File: website/docs/api-reference/configuration/manifest-json.md (hunk 5 above)
Comment (dina-kazakevich) [inline website/docs/api-reference/configuration/manifest-json.md:161]:
```suggestion
| `action`  | A user action such as linking to another page, calling an API, etc. Your component will receive a function returning a promise. The promise will return when all of the associated actions have been completed. |
```

**Pull request merged**